di-ipv-queue-stub/enqueue-event/src/services/queueService.ts

```diff
@@ -0,0 +1,112 @@
+import {
+  AddPermissionCommand,
+  CreateQueueCommand,
+  DeleteMessageCommand,
+  GetQueueUrlCommand,
+  Message,
+  QueueDoesNotExist,
+  ReceiveMessageCommand,
+  SendMessageCommand,
+  SQSClient,
+} from "@aws-sdk/client-sqs";
+
+const sqsClient = new SQSClient({ region: "eu-west-2" });
+
+export const getOrCreateSqsQueueUrl = async (
+  accountId: string,
+  queueName: string,
+): Promise<string> => {
+    try {
+        const queueUrlResponse = await sqsClient.send(new GetQueueUrlCommand({
+            QueueName: queueName,
+        }));
+        console.info(`Queue already exists ${queueName}`);
+        return queueUrlResponse.QueueUrl!;
+    } catch (error) {
+        if (error instanceof QueueDoesNotExist) {
+            console.info(`Creating queue ${queueName}`);
+
+            // Create the queue if it does not exist
+            const redrivePolicy = {
+                deadLetterTargetArn: `arn:aws:sqs:eu-west-2:${accountId}:stubQueue_F2FDLQ`,
+                maxReceiveCount: "1"
+            };
+            const createQueueResponse = await sqsClient.send(new CreateQueueCommand({
+                QueueName: queueName,
+                Attributes: {
+                    KmsMasterKeyId: "alias/sqs/QueuesKmsKey",
+                    RedrivePolicy: JSON.stringify(redrivePolicy),
+                    VisibilityTimeout: "360"
+                }
+            }));
+
+            console.info(`Created queue ${queueName}`);
+
+            // Add permissions to consume the queue from IPV Core accounts
+            await sqsClient.send(new AddPermissionCommand({
+                QueueUrl: createQueueResponse.QueueUrl,
+                Label: "add read and delete from dev, build, staging, prod perms",
+                AWSAccountIds: [
+                    "457601271792",
+                    "130355686670",
+                    "175872367215",
+                    "335257547869",
+                    "991138514218",
+                    "075701497069"
```

Comment:
I know this was copied from before but do these perms really need adding for int/prod?

Reply:
Fair point - doesn't seem likely we'll ever want int/prod to subscribe to stub queue events (pretty unlikely for staging as well tbh, but more plausible)